pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +92,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
+	AuthPlugin string `json:"auth-plugin,omitempty"`
```

Comment:
So `AuthProvider` would take precedence over any other field, but its implementation would be provided those fields? I think that is reasonable.

It does change the meaning of the fields slightly. Instead of just being "* for authentication to the kubernetes cluster", they would now be "* for authentication to the k8s cluster or to a custom intermediary".

Other than figuring out naming, is this something we can punt on until we have a concrete use case? Or does the fact that this gets serialized mean we need to figure it out up front?